python/main.py

```diff
@@ -1,139 +1,182 @@
 import json
+import settings
+import ynab_client
 
-from flask import Flask, render_template, request, jsonify, redirect
+from flask import Flask, request, jsonify, redirect
 from datetime import datetime
 from dateutil.parser import parse
 from decimal import Decimal
 
 from pynYNAB.schema.budget import Transaction
 
-import settings
-import ynab_client
-
 app = Flask('Monzo to YNAB')
 app.config['DEBUG'] = settings.flask_debug
 
 if settings.sentry_dsn:
     from raven.contrib.flask import Sentry
     sentry = Sentry(app)
 
+
 @app.route('/')
 def route_index():
     return redirect("https://github.com/scottrobertson/monzo-to-ynab", code=302)
 
+
 @app.route('/webhook', methods=['POST'])
 def route_webhook():
-    expected_delta = 0
+    ynab_client.init()
     data = json.loads(request.data.decode('utf8'))
     settings.log.debug('webhook type received %s' % data['type'])
-    entities_payee_id = None
-    subcategory_id = None
-    if data['type'] == 'transaction.created':
-
-        # Sync the account so we get the latest payees
-        ynab_client.sync()
-
-        if data['data']['amount'] == 0:
-            return jsonify({'error': 'Amount is 0'} )
-
-        # Does this account exist?
-        account = ynab_client.getaccount(settings.ynab_account)
-        if account == False:
-            return jsonify({'error': 'Account not found'} )
-
-        # Work out the Payee Name
-        if data['data'].get('merchant'):
-            payee_name = data['data']['merchant']['name']
-
-            # Get the defaults for this payee based on previously imported data
-            previous_transaction = ynab_client.findPreviousTransaction(payee_name)
-            if previous_transaction is not None:
-                settings.log.debug('A previous transaction for the payee %s has been found' % payee_name)
-                entities_payee_id = previous_transaction.entities_payee.id
-                subcategory = previous_transaction.entities_subcategory
-
-                # Include the category used, as long as it's not a split category
-                if subcategory is not None:
-                    if subcategory.name != 'Split (Multiple Categories)...':
-                        settings.log.debug('We have identified the "%s" category as a good default for this payee' % subcategory.name)
-                        subcategory_id = subcategory.id
-                    else:
-                        settings.log.debug('Split category found, so we will not use that category  for %s' % payee_name)
-                else:
-                    settings.log.debug('A subcategory was not found for the previous transaction for %s' % payee_name)
-            else:
-                settings.log.debug('A previous transaction for the payee %s has not been found' % payee_name)
+    if data.get('type') == 'transaction.created':
+        create_transaction(data['data'], settings, 0)
```

Comment:
Oh wait, it's because the `create_transaction` may return another message (if it fails). This would render "successful" even if the `create_transaction` returned the following:

```
return jsonify({'error': 'Skipping due to duplicate transaction'})
```